server/container/util-java8/src/main/java/org/apache/james/util/OptionalUtils.java

```diff
@@ -43,7 +44,17 @@
 
     @SafeVarargs
     public static <T> Optional<T> or(Optional<T>... optionals) {
-        return Arrays.stream(optionals)
+        return orStream(Arrays.stream(optionals));
+    }
+
+    @SafeVarargs
+    public static <T> Optional<T> orSuppliers(Supplier<Optional<T>>... suppliers) {
```

Comment:
`or` would be enough as Supplier is already the argument type

Reply:
Well, the distinction can not be done (even when enforcing type parameter) when there is no type parameter.

That is a testing edge case that I would like not to remove. Hence I propose to keep the naming distinction.